components/sup/doc/http_gateway_services_schema.json

```diff
@@ -0,0 +1,521 @@
+        },
+        "version": {
+          "type": "string",
+          "description": "The package version"
+        }
+      }
+    }
+  },
+  "items": {
+    "type": "object",
+    "description": "The services that the Habitat supervisor has loaded",
+    "required": [
+      "all_pkg_binds",
+      "binding_mode",
+      "binds",
+      "bldr_url",
+      "cfg",
+      "channel",
+      "config_from",
+      "desired_state",
+      "health_check",
+      "hooks",
+      "initialized",
+      "last_election_status",
+      "manager_fs_cfg",
+      "needs_reconfiguration",
+      "needs_reload",
+      "pkg",
+      "process",
+      "service_group",
+      "smoke_check",
+      "spec_file",
+      "spec_ident",
+      "sys",
+      "topology",
+      "update_strategy",
+      "user_config_updated"
+    ],
+    "properties": {
+      "all_pkg_binds": {
+        "type": "array",
+        "description": "All of the package bind information",
+        "items": {
+          "type": "object",
+          "description": "A package bind",
+          "required": [
+            "exports",
+            "service"
+          ],
+          "properties": {
+            "exports": {
+              "type": "array",
+              "description": "The binds that the service exports",
+              "items": {
+                "type": "string",
+                "description": "The bind value"
+              }
+            },
+            "service": {
+              "type": "string",
+              "description": "The bind name"
+            }
+          }
+        }
+      },
+      "binding_mode": {
+        "type": "string",
+        "description": "What type of bind this is, either strict or relaxed"
+      },
+      "binds": {
+        "type": "array",
+        "description": "Mapping of bind name to service group",
+        "items": {
+          "type": "string",
+          "description": "A colon separated list of two items: the bind name, and service group",
+          "examples": [
+            "router:builder-router.default"
+          ]
+        }
+      },
+      "bldr_url": {
+        "type": "string",
+        "description": "The URL for the Habitat Builder service"
+      },
+      "cfg": {
+        "type": "object",
+        "description": "The current configuration of this service",
+        "additionalProperties": {
+          "type": ["boolean", "string", "integer", "array", "object"]
+        }
+      },
+      "channel": {
+        "type": "string",
+        "description": "The channel this service updates itself from"
+      },
+      "composite": {
+        "type": ["null", "string"],
+        "description": "The name of the composite this service is a part of"
+      },
+      "config_from": {
+        "type": ["null", "string"],
+        "description": "The path where the config is loaded from"
+      },
+      "desired_state": {
+        "type": "string",
+        "description": "The desired state for this service, either Up or Down"
```

Comment:
Enumerate values.

Reply:
Sorry, I didn't even know this was a thing you could do when I wrote this.  I'll fix all these.